src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/Razor/ExtractToComponentCodeActionProviderTest.cs

```diff
@@ -292,6 +343,51 @@ public Task Handle_MultiPointSelection_StartSelfClosing()
             Welcome to your new app.
             """);
 
+    [Fact]
+    public Task Handle_MultipointSelection_CodeBlock()
+        => TestAsync("""
+            {|result:{|selection:<p>Hello</p>
+
+            @code {
+              |}
+            }|}
+            """);
+
+    [Fact]
+    public Task Handle_MultipointSelection_IfBlock()
+        => TestAsync("""
+            {|result:{|selection:<p>Hello</p>
+
+            @if (true) {
+              |}
+            }|}
+            """);
+
+    [Fact]
+    public Task Handle_MultipointSelection_EmbeddedIfBlock()
+        => TestAsync("""
+            {|result:{|selection:<p>Hello</p>
+
+            <div>
+                <div>
+                    @if (true) {
+                      |}
+                    }
+                </div>
+            </div>|}
+            """);
+
+    [Fact]
+    public Task Handle_MultipointSelection_CSharpBlock()
+        => TestAsync(
+            """
+            {|result:<div{|selection:>blah</div>
+
+            @{
+                RenderFragment fragment = @<Component1 Id="Comp1" Caption="Title">|] </Component1>;
```

Comment:
Should the "|]" here be "|}"? I can't see where the selection ends here

Reply:
It should, I'm surprised this passed...

Reply:
I think the parser actually just fixes my mistake by assuming I meant `|}`. Either way fixing because that's weird